### ISSUE TITLE
JOSS: PINN tests do not seem to do much

### ISSUE THREAD

**Issue by jgostick** [EMSL-Computing/Pore2Chip#19]
When running the pinn tests I get:

```
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=1.000e-03
   Epoch=100	 loss=2.893e-04
PINN training done!
   Best Epoch = 70	Best Loss = 5.960e-07
.PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=3.553e-14
   Epoch=100	 loss=1.027e-11
PINN training done!
   Best Epoch = 0	Best Loss = 3.553e-14
.PINN training started...
   Epoch=0	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
..PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
.PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
.PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
.PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
.PINN training started...
   Epoch=0	 loss=0.000e+00
   Epoch=100	 loss=0.000e+00
PINN training done!
   Best Epoch = 0	Best Loss = 0.000e+00
.
----------------------------------------------------------------------
Ran 9 tests in 0.743s

```

Most of these tests are giving losses of 0, and doing 0 epochs.

**Comment by maruti-iitm** [comment]
Hi @jgostick 

We (@mlmamud and @maruti-iitm) have updated the `setUp()` method to ensure that the initial weights are distinct from the collocation points. This avoids degenerate cases and ensures that the test loss function evaluates meaningful discrepancies during training. Please run the `test_train_pinn.py` script to see the updates.
